src/components/PortableText/PortableText.tsx

```diff
@@ -181,12 +184,13 @@ export const basePortableTextComponents: PortableTextComponents = {
   },
 };
 
-export const BasePortableTextProvider: FC<{
-  children?: React.ReactNode;
-}> = (props) => {
+export function PortableTextWithDefaults<
+  B extends TypedObject = PortableTextBlock,
+>(props: PortableTextProps<B>): React.JSX.Element {
   return (
-    <PortableTextComponentsProvider components={basePortableTextComponents}>
-      {props.children}
-    </PortableTextComponentsProvider>
+    <PortableText
+      {...props}
+      components={merge(basePortableTextComponents, props.components)}
```

Comment:
Is this merging the components deep enough?

`_.merge` docs say:
> Array and plain object properties are merged recursively.

I thought there was `_.mergeDeep` but I can't find it, so maybe lodash has always just had that behaviour

Reply:
Ah I see this PR came after we spoke about regressions, so guessing it's fine and this was the fix

Reply:
Yep i think that's right! Will come back to this PR